src/vcpkg/install.cpp

```diff
@@ -621,11 +621,53 @@ namespace vcpkg::Install
         }
     }
 
-    CMakeUsageInfo get_cmake_usage(const Filesystem& fs, const InstalledPaths& installed, const BinaryParagraph& bpgh)
+    static constexpr StringLiteral ADD_LIBRARY_CALL = "add_library(";
+
+    static const char* find_add_library(const char* real_first, const char* first, const char* last)
+    {
+        for (;;)
+        {
+            first = std::search(first, last, ADD_LIBRARY_CALL.begin(), ADD_LIBRARY_CALL.end());
+            if (first == real_first || first == last || !Parse::ParserBase::is_word_char(*(first - 1)))
+            {
+                return first;
```

Comment:
The caller skips over ADD_LIBRARY_CALL.size(), should that be here so that only this function need see the variable ADD_LIBRARY_CALL?